lib/sensors/accelerometer.cpp

```diff
@@ -0,0 +1,167 @@
+std::optional<core::RawAccelerationData> Accelerometer::read()
+{
+  /* check to see if the values are ready to be read */
+  auto dataReady = i2c_.readByte(mux_address_, kDataReady);
+  if (!dataReady) {
+    log_.log(core::LogLevel::kFatal, "acceleration data could not be read");
+    return std::nullopt;
+  }
+  if (dataReady.value() == 0x00) {
+    log_.log(core::LogLevel::kFatal, "acceleration data not ready yet to be read");
+    return std::nullopt;
+  }
+
+  // x axis
+  auto resultX = getRawAccelerationX();
+  if (!resultX) return std::nullopt;
+  const std::int16_t XRawAcc
+    = resultX.value() >> 2; /* shifted by 2 as 14bit resolution is used in high performance mode */
+  core::Float XAcceleration = static_cast<core::Float>(XRawAcc);
+  XAcceleration             = XAcceleration / 1000; /* mg to g */
+  XAcceleration = XAcceleration * 1.952; /* Multiply with sensitivity 1.952 in high performance
+                                            mode, 14bit, and full scale +-16g */
+
+  // y axis
+  auto resultY = getRawAccelerationY();
+  if (!resultY) return std::nullopt;
+  const std::int16_t YRawAcc = resultY.value() >> 2;
+  core::Float YAcceleration  = static_cast<core::Float>(YRawAcc);
+  YAcceleration              = YAcceleration / 1000;
+  YAcceleration              = (YAcceleration * 1.952);
+
+  // z axis
+  auto resultZ = getRawAccelerationZ();
+  if (!resultZ) return std::nullopt;
+  const std::int16_t ZRawAcc = resultZ.value() >> 2;
+  core::Float ZAcceleration  = static_cast<core::Float>(ZRawAcc);
+  ZAcceleration              = ZAcceleration / 1000;
+  ZAcceleration              = ZAcceleration * 1.952;
+
+  const std::optional<core::RawAccelerationData> rawAcceleration{
+    std::in_place, XAcceleration, YAcceleration, ZAcceleration, std::chrono::system_clock::now()};
+
+  return rawAcceleration;
+}
+
+core::Result Accelerometer::configure()
+{
+  // if device id is incorrect, then the address received is wrong
+  const auto deviceID = i2c_.readByte(mux_address_, kDevId);
+  if (!deviceID) {
+    log_.log(core::LogLevel::kFatal, "Failure to read device id of accelerometer");
+    return core::Result::kFailure;
+  }
+  if (deviceID.value() != expectedDevId) {
+    log_.log(core::LogLevel::kFatal, "Failiure: Accelerometer didn't give correct device id");
+    return core::Result::kFailure;
+  }
+
+  // configure the sensor according to the following settings :
+
+  /* Sampling rate of 200 Hz */
+  /* Enable high performance mode */
+  const core::Result ctrl1Result = i2c_.writeByteToRegister(mux_address_, kCTRL1, 0x64);
+  if (ctrl1Result == core::Result::kFailure) return core::Result::kFailure;
+
+  /* Enable block data update */
+  /* Enable address auto increment */
+  const core::Result ctrl2Result = i2c_.writeByteToRegister(mux_address_, kCTRL2, 0xC);
+  if (ctrl2Result == core::Result::kFailure) return core::Result::kFailure;
+
+  /* Full scale +-16g */
+  /* Filter bandwidth = ODR/2 */
```

Comment:
stick with `//` comments here; `/* ... */` is usually more for docs in the header (no clear rule, just preference)